diff --git a/src/crypto/x509/root.go b/src/crypto/x509/root.go
index cc53f7aefc..b29e6cfe01 100644
--- a/src/crypto/x509/root.go
+++ b/src/crypto/x509/root.go
@@ -10,7 +10,11 @@ package x509
 // and run "go generate". See https://golang.org/issue/38843.
 //go:generate go run root_ios_gen.go -version 55188.40.9
 
-import "sync"
+import (
+	"io/ioutil"
+	"os"
+	"sync"
+)
 
 var (
 	once           sync.Once
@@ -24,8 +28,20 @@ func systemRootsPool() *CertPool {
 }
 
 func initSystemRoots() {
-	systemRoots, systemRootsErr = loadSystemRoots()
-	if systemRootsErr != nil {
-		systemRoots = nil
+	if os.Getenv("GO_USE_SYSTEM_ROOTS") == "1" {
+		systemRoots, systemRootsErr = loadSystemRoots()
+		if systemRootsErr != nil {
+			systemRoots = nil
+		}
+	}
+
+	if file := os.Getenv("NIX_SSL_CERT_FILE"); file != "" {
+		data, err := ioutil.ReadFile(file)
+		if err == nil {
+			if systemRoots == nil {
+				systemRoots = NewCertPool()
+			}
+			systemRoots.AppendCertsFromPEM(data)
+		}
 	}
 }
